src/pages/dashboard.tsx

```diff
@@ -0,0 +1,129 @@
+  | "REJECTED"
+  | "WAITLISTED"
+  | "DECLINED";
+
+const ApplicationStatus = ({ status }: { status: ApplicationStatusType }) => {
+  const parsedStatuses = {
+    IN_PROGRESS: "Application In Progress...",
+    PENDING_REVIEW: "Application Submitted!",
+    IN_REVIEW: "Application In Review",
+    ACCEPTED: "Application Accepted! 🥳",
+    REJECTED: "Application Not Accepted",
+    WAITLISTED: "Waitlisted",
+    DECLINED: "You're Not Coming :(",
+  };
+
+  const statusClassName = {
+    IN_PROGRESS: "bg-primary-200 text-violet-500",
+    PENDING_REVIEW: "bg-primary-300 border border-primary-600 text-primary-600",
+    IN_REVIEW: "bg-primary-300 border border-primary-600 text-primary-600",
+    ACCEPTED: "bg-primary-500 text-primary-100",
+    REJECTED: "bg-violet-500 text-primary-100",
+    WAITLISTED: "bg-violet-500 text-primary-100",
+    DECLINED: "bg-violet-500 text-primary-100",
+  };
+
+  return (
+    <div
+      className={`text-xl font-medium capitalize ${statusClassName?.[status ?? "IN_PROGRESS"]} mx-auto flex w-max select-none justify-center rounded-full px-10 py-2.5 text-center`}
+    >
+      {parsedStatuses?.[status ?? "IN_PROGRESS"]}
+    </div>
+  );
+};
+
+const buttonStatus = (status: ApplicationStatusType) => {
+  const parsedStatuses = {
+    IN_PROGRESS: "Continue Your Application",
+    PENDING_REVIEW: "View Application",
+    IN_REVIEW: "View Application",
+    ACCEPTED: "RSVP by XXXX",
+    REJECTED: "",
+    WAITLISTED: "View Application",
+    DECLINED: "Change RSVP",
+  };
+
+  return parsedStatuses?.[status ?? "IN_PROGRESS"];
+};
+
+const Dashboard = () => {
+  const { data: application } = api.application.get.useQuery();
+  const router = useRouter();
+
+  const onButtonClick = (status: ApplicationStatusType) => {
+    switch (status) {
+      case "IN_PROGRESS":
+        void router.push("/apply?step=persona");
+      case "PENDING_REVIEW":
+        void router.push("/apply?step=review");
+      default:
+        void router.push("/apply?step=review");
+    }
+  };
+
+  return (
+    <>
+      <Head>
+        <title>Hack Western</title>
+        <meta
+          name="description"
+          content="Hack Western: One of Canada's largest annual student-run hackathons based out of Western University in London, Ontario."
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main className="flex h-svh max-h-svh flex-col items-center bg-primary-50">
+        <ApplyNavbar />
+        <div className="flex w-full flex-1 flex-grow items-center">
+          <div
+            id="left-panel"
+            className="lg:w-xl flex h-full flex-grow flex-col justify-center gap-4 bg-primary-100 p-9 pt-12 text-center lg:max-w-xl"
+          >
+            <div className="pb-2.5 text-3xl font-bold text-slate-700">
+              {
+                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                `Welcome Back, ${application?.firstName || "Hacker"}!`
```

Comment:
need to not use `??` because `firstName` could be empty string and it would nullish coalesce to empty string